src/prefect/backend/flow_run.py

```diff
@@ -704,14 +727,14 @@ def get_task_run(
                 # just be one run id but for mapped tasks there will be multiple
                 for task_run_id in task_run_ids:
                     result = self._cached_task_runs[task_run_id]
-                    if result.map_index == -1:
+                    if result.map_index == map_index:
```

Comment:
might be worth raising an error somewhere in this function if `map_index != -1` and `task_slug` is still `None` (or implement logic for other identifiers)

Reply:
Yeah I'm thinking of tearing out lookup by `task` object and adding that check in a separate PR; it feels kind of weird as is.